### PR TITLE
aliae: init at 0.22.1

### PR DESCRIPTION
**Pull request by vedantmgoyal9**
## Description of changes

Aliae is a CLI-based tool by the creator of the famous [oh-my-posh](https://ohmyposh.dev) to manage shells (bash, zsh, fish).

GitHub: https://github.com/jandedobbeleer/aliae
Homepage: https://aliae.dev

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/al/aliae/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  tags = [
+    "netgo"
+    "osusergo"
+    "static_build"
+  ];
+
+  postInstall = ''
+    mv $out/bin/{src,aliae}
+    installShellCompletion --cmd aliae \
```
Comment (sikmir) [inline pkgs/by-name/al/aliae/package.nix:40]:
```suggestion
  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd aliae \
```

File: pkgs/by-name/al/aliae/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```
Comment (sikmir) [inline pkgs/by-name/al/aliae/package.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:
```

File: pkgs/by-name/al/aliae/package.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  installShellFiles,
+}:
+
+buildGoModule rec {
+  pname = "aliae";
+  version = "0.22.1";
+
+  src = fetchFromGitHub {
+    owner = "jandedobbeleer";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/by-name/al/aliae/package.nix:15]:
```suggestion
    repo = "aliae";
```

File: pkgs/by-name/al/aliae/package.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      mv $out/bin/{src,aliae}
+    ''
+    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      installShellCompletion --cmd aliae \
+        --bash <($out/bin/aliae completion bash) \
+        --fish <($out/bin/aliae completion fish) \
+        --zsh <($out/bin/aliae completion zsh)
+    '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/al/aliae/package.nix:49]:
```suggestion
  meta = {
```

File: pkgs/by-name/al/aliae/package.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    description = "Cross shell and platform alias management";
+    mainProgram = "aliae";
+    homepage = "https://aliae.dev";
+    changelog = "https://github.com/JanDeDobbeleer/aliae/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ vedantmgoyal9 ];
```
Comment (eclairevoyant) [inline pkgs/by-name/al/aliae/package.nix:55]:
```suggestion
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ vedantmgoyal9 ];
```

File: pkgs/by-name/al/aliae/package.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.Version=${version}"
+  ];
+
+  tags = [
+    "netgo"
+    "osusergo"
+    "static_build"
```
Comment (emilazy) [inline pkgs/by-name/al/aliae/package.nix:35]: I’m not sure if we want this. Does it break without?

Comment (vedantmgoyal9) [reply pkgs/by-name/al/aliae/package.nix:35]: I copied it from https://github.com/JanDeDobbeleer/aliae/blob/main/src/.goreleaser.yml

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 339118` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>aliae</li>
  </ul>
</details>

**Comment by emilazy** [comment]
Thanks!

**Pull request merged**